aggregator_core/src/datastore.rs

```diff
@@ -6045,3 +6145,21 @@ impl From<aws_lc_rs::error::Unspecified> for Error {
         Error::Crypt
     }
 }
+
+/// Details of a [`Error::TimeUnaligned`] error.
+#[derive(Debug)]
+pub struct TimeUnaligned {
```

Comment:
Another option, which would let you define this inline in the error enum and save you the `Display` impl, would be:

```
pub enum Error {
    ...
    #[error("time is unaligned (precision = {1}, inner error = {2}")
    TimeUnaligned {
        task_id: TaskId,
        time_precision: Duration,
        inner_error: Box<janus_messages::Error>,
    }
    ...
}
```

Reply:
Thanks for this. I first tried to do this, but I got caught up, believe it or not, interacting with the enum variant's internal fields during the matching in the `http_handlers.rs`, which you helpfully elucidated above. Very nice.